adagrams/game.py

```diff
@@ -1,11 +1,127 @@
+import random
+
 def draw_letters():
-    pass
+
+    LETTER_POOL = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    
+    # initiate draw_letters to record and return drawn random letters
+    # initiate letter_freq to check the frequency of drawn out random letters
+    draw_letters_list = []
+    letter_freq = {}
+    
+    # create random letter picking from the letter poll
+    # only append letters that are no bigger than letter poll frequency
+    while len(draw_letters_list) < 10:
+        cur_random_letter = random.choice(list(LETTER_POOL.keys()))
+        if cur_random_letter in letter_freq.keys():
+            letter_freq[cur_random_letter] += 1
+        else:
+            letter_freq[cur_random_letter] = 1
+        if letter_freq[cur_random_letter] > LETTER_POOL[cur_random_letter]:
+            continue
+        else:
+            draw_letters_list. append(cur_random_letter)
+    return draw_letters_list
 
 def uses_available_letters(word, letter_bank):
-    pass
+    
+    # check each letter's frequency and record in letter_bank_freq
+    letter_bank_freq = {}
+    for letter in letter_bank:
+        if letter in letter_bank_freq:
+            letter_bank_freq[letter] += 1
+        else:
+            letter_bank_freq[letter] = 1
+
+    # check if every letter or its lower-case form in word existed in letter_bank 
+    # also check the occurance of each letter not bigger than its frequency in the letter_bank      
+    for letter in word:
+        if letter and letter.upper() not in letter_bank:
```

Comment:
Instead of checking letter and letter.upper, you can capitalize all the letters in `word` on line 65. For example: 
```python
for letter in word.upper():
```

Also, notice you have a for loop and then you use the `in` operator on line 66 so we have quadratic time complexity in this method.